xdem/spatialstats.py

```diff
@@ -826,20 +816,19 @@ def infer_heteroscedasticity_from_stable(
         Dataframe of binned spread statistic with explanatory variables,
         Error function with explanatory variables
     """
-
     # Create placeholder variables names if those don't exist
     if list_var_names is None:
         list_var_names = ["var" + str(i + 1) for i in range(len(list_var))]
 
     # Get the arrays for proxy values and explanatory variables
     list_all_arr, gsd = _preprocess_values_with_mask_to_array(
-        values=[dvalues] + list_var, include_mask=stable_mask, exclude_mask=unstable_mask, preserve_shape=False
+        values=[dvalues, list_var], include_mask=stable_mask, exclude_mask=unstable_mask, preserve_shape=False,
```

Comment:
This modifications triggered an issue : it creates a nested list where `list_var` is treated as a single element, resulting in [dvalues, [list_var]].